pkg/daemon/ceph/agent/flexvolume/controller.go

```diff
@@ -301,12 +302,29 @@ func (c *Controller) GetAttachInfoFromMountDir(mountDir string, attachOptions *A
 			return fmt.Errorf("failed to get pods in namespace %s: %+v", attachOptions.PodNamespace, err)
 		}
 
-		pod := findPodByID(pods, types.UID(attachOptions.PodID))
+		pod = findPodByID(pods, types.UID(attachOptions.PodID))
 		if pod != nil {
 			attachOptions.Pod = pod.GetName()
 		}
 	}
 
+	if attachOptions.PodUser == nil || attachOptions.PodGroup == nil {
+		if pod == nil {
```

Comment:
when would `pod` be `nil`? Wouldn't it be found above on 305? If so, we could get the pod user and group above as well.